### PR TITLE
Windows: Tidy verifyContainerSettings

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

This fixes up some more TODO Windows. In particular this tidies up verifyContainerSettings into platform agnostic and platform specific parts. It also godocs some functions I changed while in that area of the code.

### CODE REVIEW

**Comment by runcom** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

File: daemon/daemon.go [hunk 1]
```diff
@@ -967,3 +968,35 @@ func getDefaultRouteMtu() (int, error) {
 	}
 	return 0, errNoDefaultRoute
 }
+
+// verifyContainerSettings performs validation of the hostconfig and config
+// structures.
+func (daemon *Daemon) verifyContainerSettings(hostConfig *runconfig.HostConfig, config *runconfig.Config) ([]string, error) {
+
+	// First perform verification of settings common across all platforms.
```
Comment (tiborvass) [inline daemon/daemon.go]: @jhowardmsft why did you remove the 3-line comment? Did something change?

Comment (jhowardmsft) [inline daemon/daemon.go]: I really don't think it's necessary anymore with the platform specific factoring out.

Comment (tiborvass) [inline daemon/daemon.go]: Ok thanks

Comment (jhowardmsft) [inline daemon/daemon.go]: Oh hang on, are you talking about the same PR? I thought it was a different one I removed the comment from.

Comment (tiborvass) [inline daemon/daemon.go]: Same PR yeah: https://github.com/docker/docker/pull/15179/files#diff-b56258ecab11f7b797306907327cc904L134

Comment (jhowardmsft) [inline daemon/daemon.go]: Ah, another 3-line comment :) I'd probably say this is OK. In the context of the code as it stands now, I would say it's obvious the check is being done on the daemon rather than the client. I added that comment way back when I did the original move from client to server. It's truly redundant now and I probably should never have put it in in the first place

**Comment by jhowardmsft** [comment]
The Windows CI failures for /dev/urandom are common across multiple PRs this week. Not sure about the other failure but doesn't look to be related.

**Comment by tiborvass** [comment]
@jhowardmsft Can you repro the following error locally even if independent from PR?

```
FAIL: docker_cli_run_test.go:159: DockerSuite.TestRunWorkingDirectory

docker_cli_run_test.go:170:
    c.Errorf("--workdir failed to set working directory")
... Error: --workdir failed to set working directory
```

**Comment by jhowardmsft** [comment]
Yes :) Exactly the same error. From a Windows CLI on master to an Ubuntu daemon running master.

**Comment by tiborvass** [comment]
Opened #15192
